otherlibs/runtime_events/runtime_events.mli

```diff
@@ -119,6 +119,56 @@ module Timestamp : sig
     val to_int64 : t -> int64
 end
 
+module Type : sig
+  type 'a t
+  (** The type for an user event content type *)
+
+  val event : unit t
+  (** An event has no data associated with it *)
+
+  type span = Begin | End
+
+  val span : span t
+  (** An event that have a beginning and an end *)
+
+  val counter : int t
+  (** A counter is a number *)
+
+  val register : encode:(bytes -> 'a -> int) -> decode:(bytes -> int -> 'a)
+                                                                        -> 'a t
+  (** Registers a custom type by providing an encoder and a decoder. The encoded
+      writes the value in the provided buffer and returns  *)
```

Comment:
is this sentence unfinished? :)

Also, it's clear to me that `encode` returns how many bytes it wrote, but does `decode` take the same length (i.e. a slice, in essence), or an offset? I think this could use more doc.

Reply:
Oops. Indeed this should be better documented now that `encode` and `decode` are a bit more complex.
`decode` is given a slice (as a consequence the encoder can be of variable length), and has to return the value.